src/main/java/appeng/block/crafting/ICraftingUnitType.java

```diff
@@ -0,0 +1,46 @@
+package appeng.block.crafting;
+
+import net.minecraft.world.item.Item;
+
+/**
+ * Implemented by classes/enums meant to provide their own types of crafting CPU blocks.
+ */
+public interface ICraftingUnitType {
+
+    /**
+     * @return the capacity of a given crafting <i>storage</i> block in bytes (should be 0 if not storage).
+     */
+    int getStorageBytes();
+
+    /**
+     * @return how many co-processors a crafting unit provides. For lag-mitigation purposes, a hard-coded limit has been
+     *         set of 16 threads for any given co-processing unit block.
+     */
+    int getAcceleratorThreads();
+
+    /**
+     * @return whether a given crafting unit type corresponds with a crafting <i>accelerator</i>. By default, anything
+     *         with a non-negative number of threads is assumed to be an accelerator.
+     */
+    default boolean isAccelerator() {
+        return this.getAcceleratorThreads() > 0;
+    };
+
+    /**
+     * @return whether a given crafting unit type corresponds with a crafting <i>storage</i> block. By default, anything
+     *         with a non-negative storage capacity in bytes is assumed to be a storage block.
+     */
+    default boolean isStorage() {
+        return this.getStorageBytes() > 0;
+    }
+
+    /**
+     * @return whether a given crafting unit type corresponds with a crafting <i>monitor</i>.
+     */
+    boolean isStatus();
```

Comment:
I don't think these checks make sense to have there, nobody should be overriding them. The `isStatus` one in particular will probably cause issues if the BE doesn't extend `CraftingStatusBlockEntity` (see my comment in `CraftingCPUCluster`).